### PR TITLE
Catalog multiall

### PR DESCRIPTION
**Pull request by techeng322**

### CODE REVIEW

File: lib/getTBAInitializeCall.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Interface } from "ethers/lib/utils"
+import erc6551AccountAbi from "@/lib/abi/erc6551-account.json"
+import { ACCOUNT_IMPLEMENTATION } from "./consts"
+
+const getTBAInitializeCall = async (tbaAddress) => {
```
Comment (sweetmantech) [inline lib/getTBAInitializeCall.tsx]: does this need to be `async`?

File: hooks/useTBAPurchase.tsx [hunk 2]
```diff
@@ -37,11 +39,21 @@ const useTBAPurchase = () => {
         zoraTotalPrice.toString(),
       ) as any
       const tba = getAccount(zoraNextTokenId)
-      const soundMintCall = await getSoundMintCall(tba)
+      const tbaInitializationCall = await getTBAInitializeCall(tba)
```
Comment (sweetmantech) [inline hooks/useTBAPurchase.tsx]: I don't think we need to `await` this 
```suggestion
      const tbaInitializationCall = getTBAInitializeCall(tba)
```

File: lib/getCosignMintCall.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { Interface } from "ethers/lib/utils"
+import cosignAbi from "@/lib/abi/catalog-cosign.json"
+import { BigNumber } from "ethers"
+import { CATALOGCOSIGN_ADDRESS, CATALOG_ID, CATALOG_REFERRAL } from "./consts"
+import getCosignPrice from "./getCosignPrice"
+
+const getCosignMintCall = async (mintRecipient) => {
+  const tokenPrice = await getCosignPrice()
```
Comment (sweetmantech) [inline lib/getCosignMintCall.tsx]: we can `hard-code` cosign price
- same on `testnet` vs `mainnet` ✅ 
- same on `any` token ✅ 
- removes `async` requirement ⚡ 
```suggestion
  const tokenPrice = 1000000000000000n
```

File: hooks/useWalletTransaction.tsx [hunk 4]
```diff
@@ -38,7 +38,7 @@ const useWalletTransaction = () => {
       return false
     } catch (error) {
       // eslint-disable-next-line no-console
-      console.error(error)
+      console.error(error, "ZIAD ERROR")
```
Comment (sweetmantech) [inline hooks/useWalletTransaction.tsx]: use `handleTxError` for `console.error` on tx handling

File: lib/getCosignPrice.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  const tokenPrice = await publicClient.readContract({
+    address: CATALOGCOSIGN_ADDRESS as `0x${string}`,
+    abi: cosignAbi,
+    functionName: "tokenPrice",
+  })
+
+  return tokenPrice
+}
+
+export default getCosignPrice
```
Comment (sweetmantech) [inline lib/getCosignPrice.tsx]: we can remove this and hard-code the price

File: lib/getSoundMintCall.tsx [hunk 6]
```diff
@@ -1,30 +1,36 @@
...
-  const { args, functionName, address: SUPERMINTER, value, abi } = mintParams.mint.input
-  const soundMintDataV2 = new Interface(abi).encodeFunctionData(functionName, args)
+    const mintSchedules = await anyPublicClient.editionV2.mintSchedules({
+      editionAddress: SPOTIFY_DROP_ADDRESS,
+    })
+    const mintParams = await anyPublicClient.editionV2.mintParameters({
+      account: mintRecipient,
+      chain: CHAIN,
+      schedule: mintSchedules.activeSchedules[0],
+      quantity,
```
Comment (sweetmantech) [inline lib/getSoundMintCall.tsx]: why are we adding a `quantity` param?

File: hooks/useTBAPurchase.tsx [hunk 7]
```diff
@@ -37,11 +39,21 @@ const useTBAPurchase = () => {
...
+      const tbaInitializationCall = getTBAInitializeCall(tba)
+      const soundQuantity = 1
+      const soundMintCall = await getSoundMintCall(tba, soundQuantity)
+      if (!soundMintCall) {
+        setLoading(false)
+        return false
+      }
       const soundMintCallValue = BigNumber.from(soundMintCall.value)
-      const totalPrice = zoraTotalPrice.add(soundMintCallValue)
+      const cosignMintCall = await getCosignMintCall(tba)
```
Comment (sweetmantech) [inline hooks/useTBAPurchase.tsx]:
```suggestion
      const cosignMintCall = getCosignMintCall(tba)
```

File: lib/getSoundMintCall.tsx (hunk 6 above)
Comment (techeng322) [reply lib/getSoundMintCall.tsx]: fixed ✅

File: lib/getCosignPrice.tsx (hunk 5 above)
Comment (techeng322) [reply lib/getCosignPrice.tsx]: fixed ✅

File: hooks/useWalletTransaction.tsx (hunk 4 above)
Comment (techeng322) [reply hooks/useWalletTransaction.tsx]: added ✅

File: lib/getCosignMintCall.tsx (hunk 3 above)
Comment (techeng322) [reply lib/getCosignMintCall.tsx]: fixed ✅

File: hooks/useTBAPurchase.tsx (hunk 2 above)
Comment (techeng322) [reply hooks/useTBAPurchase.tsx]: fixed ✅

File: lib/getTBAInitializeCall.tsx (hunk 1 above)
Comment (techeng322) [reply lib/getTBAInitializeCall.tsx]: fixed ✅

File: lib/clients/viem.tsx [hunk 8]
```diff
@@ -7,14 +7,12 @@ import {
...
 import getViemNetwork from "../viem/getViemNetwork"
-import getAlchemyRpcUrl from "../alchemy/getAlchemyRpcUrl"
 
 export const getPublicClient = (chainId: number) => {
-  const RPC_URL = getAlchemyRpcUrl(chainId)
   const chain = getViemNetwork(chainId)
   let publicClient = createPublicClient({
     chain: chain as Chain,
-    transport: http(RPC_URL),
+    transport: http(),
```
Comment (sweetmantech) [inline lib/clients/viem.tsx:15]: wow!!! 
it works without RPC_URL?

**Review by sweetmantech — APPROVED** [review]

**Pull request merged**